R/spc2mpspline.R

```diff
@@ -35,9 +37,113 @@
 #'
 setMethod("spc2mpspline", signature(object = "SoilProfileCollection"),
           function(object, var_name = NULL,
-                           pattern = "R|Cr|Cd|qm",
-                           hzdesgn = guessHzDesgnName(object),
-                           ...) {
+                   pattern = "R|Cr|Cd|qm",
+                   hzdesgn = guessHzDesgnName(object),
```

Comment:
Since this is kind of important, I'd suggest following similar pattern matching of horizon designation in other functions and use what is set via or supplied. There are some cases where you don't have horizon designations (equal-interval sampling, or perhaps synthetic data) and would want to set this to NA or NULL to "turn off" this filtering.

Reply:
I agree, ms-splines are a major tool for DSM-type approaches,  where it's very, very common to have zero hz designations.